.github/workflows/cpp-packaging.yml

```diff
@@ -605,6 +613,44 @@ jobs:
           name: firebase-cpp-sdk-${{ matrix.sdk_platform }}${{ matrix.suffix}}-package
           path: firebase-cpp-sdk-${{ matrix.sdk_platform }}${{ matrix.suffix}}-package.tgz
 
+      - name: Check for unrenamed namespaces
+        shell: bash
+        run: |
+          set +e
+          if [[ "${{ matrix.sdk_platform }}" == "darwin" ]]; then
+            nm=bin/llvm-nm
+          elif [[ "${{ matrix.sdk_platform }}" == "windows" ]]; then
+            if [[ "${{ matrix.sdk_platform }}" == *"x64"* ]]; then
+              nm="bin/nm --target pe-x86-64"
+            else
+              nm="bin/nm --target pe-i386"
+            fi
+          else  # sdk_platform = linux
+            nm=bin/nm
+          fi
+          ${nm} `find firebase-cpp-sdk-*-package/libs/${{ matrix.sdk_platform }} -type f` | grep '^[0-9a-fA-F][0-9a-fA-F]* ' | cut -d' ' -f3- > raw_symbols.txt
+          if [[ "${{ matrix.sdk_platform }}" == "windows" ]]; then
+            cat raw_symbols.txt | sort | uniq | bin/demumble | grep '^[a-zA-Z_]*::' > cpp_symbols.txt
+          elif [[ "${{ matrix.sdk_platform }}" == "darwin" ]]; then
+            # remove leading _ on mach-o symbols
+            cat raw_symbols.txt | sort | uniq | sed 's/^_//' | bin/c++filt | grep '^[a-zA-Z_]*::' > cpp_symbols.txt
+          else  # linux
+            cat raw_symbols.txt | sort | uniq | bin/c++filt | grep '^[a-zA-Z_]*::' > cpp_symbols.txt
+          fi
+          # cpp_symbols.txt contains a list of all C++ symbols, sorted and deduped
+          # get a list of all top-level namespaces (except system namespaces)
+          cat cpp_symbols.txt | cut -d: -f1 | sort | uniq > cpp_namespaces.txt
+          echo "List of all namespaces found:"
+          cat cpp_namespaces.txt
+          # Filter out renamed namespaces (f_b_*), standard namespaces, and the public namespace (firebase::)
+          allow_namespaces="f_b_.*|std|stdext|__gnu_cxx|cxxabiv1|firebase"
```

Comment:
Thought: could this list be in a configuration file checked into the repo so it's not quite as hidden deep into a build script?